src/main/java/frc/team2412/robot/subsystem/IndexSubsystem.java

```diff
@@ -1,11 +1,189 @@
 package frc.team2412.robot.subsystem;
 
-import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import static frc.team2412.robot.subsystem.IndexSubsystem.IndexConstants.*;
+import static frc.team2412.robot.subsystem.IndexSubsystem.IndexConstants.IndexMotorState.*;
+import static frc.team2412.robot.subsystem.IndexSubsystem.IndexConstants.IndexPositionState.*;
 
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.util.MultiplexedColorSensor;
+
+import com.revrobotics.ColorMatch;
 
 public class IndexSubsystem extends SubsystemBase {
-    public IndexSubsystem(WPI_TalonFX indexMotor) {
+
+    // Constants
+
+    public static class IndexConstants {
+
+        public static Alliance teamColor = DriverStation.getAlliance();
+
+        // Color Sensor values
+
+        public static Color BLUE_CARGO_COLOR = new Color(0, 0, 1);
+        public static Color RED_CARGO_COLOR = new Color(1, 0, 0);
+
+        // Index Motor Speeds
+
+        public static final double INDEX_IN_SPEED = 0.5; // will change values later
+        public static final double INDEX_OUT_SPEED = -0.5; // will also change later
+
+        // Index Motor States
+
+        public static enum IndexMotorState {
+            IN, OUT, STOPPED;
+        }
+
+        // Position States
+
+        public static enum IndexPositionState {
+            HAS_BALL, HAS_NO_BALL;
+        }
+
+        // The current limit
+        public static final SupplyCurrentLimitConfiguration MAX_MOTOR_CURRENT = new SupplyCurrentLimitConfiguration(
+                true, 40, 40, 500);
 
     }
+
+    // New ColorMatching Instance
+
+    private ColorMatch colorMatcher = new ColorMatch();
+
+    // Define Hardware
+
+    private final MultiplexedColorSensor ingestColorSensor;
+    private final MultiplexedColorSensor feederColorSensor;
+
+    private final WPI_TalonFX ingestMotor;
+    private final WPI_TalonFX feederMotor;
+
+    // States
+
+    private IndexMotorState ingestMotorState;
+    private IndexMotorState feederMotorState;
+
+    private IndexPositionState ingestBallState;
+    private IndexPositionState feederBallState;
+
+    // Constructor
+
+    public IndexSubsystem(WPI_TalonFX firstMotor, WPI_TalonFX secondMotor, MultiplexedColorSensor firstColorSensor,
+            MultiplexedColorSensor secondColorSensor) {
+        this.ingestMotor = firstMotor;
+        this.feederMotor = secondMotor;
+        this.ingestColorSensor = firstColorSensor;
+        this.feederColorSensor = secondColorSensor;
+
+        this.ingestMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+        this.feederMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
```

Comment:
Please first reset the motor configs to their default state, then put them in brake mode before configuring the current limit.